pkg/storegateway/indexheader/stream_binary_reader.go

```diff
@@ -148,7 +175,107 @@ func newFileStreamBinaryReader(path string, postingOffsetsInMemSampling int, log
 		return nil, err
 	}
 
-	return r, nil
+	return r, err
+}
+
+// loadSparseFromDisk loads in sparse index-header from disk.
+func (r *StreamBinaryReader) loadSparseFromDisk(ctx context.Context, logger log.Logger, sparseHeadersPath string, sparseData []byte, postingOffsetsInMemSampling int) (err error) {
+	span, _ := tracing.StartSpan(ctx, "index_header_load_sparse_header")
```

Comment:
Optional: If you use the `spanlogger` like this you can use the same thing for logging and the span: https://github.com/grafana/mimir/blob/12792f6adf17cb36e9aa088f43ec58aea82e560c/pkg/distributor/query.go#L202-L220